### PR TITLE
Scrabble Sinatra Completed

### PR DESCRIPTION
**Pull request by carjug**
We completed our Scrabble Sinatra project through the Lunch requirements. Nothing is formatted so sorry about the gross default HTML layout. We probably could have come up with better method and variable names but we didn't have time, so please let us know if anything isn't clear.

Your feedback is valuable!!

<3 Shanna and Carly

### CODE REVIEW

File: lib/score.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+         # gets the score value for the letter and adds it to the total word score
+         letter_score = score_letter(letter)
+         total_word_score += letter_score
+       end
+
+       return total_word_score
+     end
+
+
+     def self.score_word_by_letter(word)
```
Comment (kariabancroft) [inline lib/score.rb:47]: As a general rule, you can initialize an array by using `[]`, without the space `[ ]`.

File: lib/score.rb [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      # final_array = [["c", 3], ["a", 1], ["t", 1]]
+      final_array = [ ]
+
+      letters = word.upcase.split("")
+
+      letters.each do |letter|
+        inside_array = [ ]
+        inside_array.push(letter)
+        inside_array.push(self.score_letter(letter))
+        final_array.push(inside_array)
```
Comment (kariabancroft) [inline lib/score.rb:57]: I like the way you've used a local variable to store the `inside_array` to make this clear.

File: lib/score.rb [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+     def self.score_array_of_words(array)
+      # final_array = [["candy", 11], ["apple", 9], ["whatever, 17"]]
+      final_array = [ ]
+
+      array.each do |word|
+        inside_array = [ ]
+        inside_array.push(word)
+        inside_array.push(self.score_word(word))
+        final_array.push(inside_array)
+      end
```
Comment (kariabancroft) [inline lib/score.rb:73]: Now that we know about blocks & procs, lines 53-58 look a lot like 68-73. Maybe a good place to explore!

File: lib/score.rb [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+   def self.score_letters_in_array(array)
+    final_array = []
+
+    array.each do |word|
+      final_array.push(self.score_word_by_letter(word))
+    end
+    return final_array
+   end
+
+    #  binding.pry
```
Comment (kariabancroft) [inline lib/score.rb:88]: Try to clean up this type of debugging work before you submit your PR

File: scrabble_sinatra.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
+require 'sinatra'
+require 'sinatra/reloader'
+require './lib/score.rb'
+
+class ScrabbleSinatra < Sinatra::Base
```
Comment (kariabancroft) [inline scrabble_sinatra.rb:5]: Your routes and HTTP verbs are logically set up - nice job

File: views/layout.erb [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<html>
+<head>
+  <title>Scrabble Sinatra</title>
```
Comment (kariabancroft) [inline views/layout.erb:3]: Excellent!

File: views/home.erb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<html>
+<head>
+  <title>Home Page</title>
+</head>
+<body>
+  <h1>SCRABBLE!!!</h1>
+    <p>
+      Go To the <strong>Score</strong> page to score one word or the <strong>Score Multiple Words</strong> page to score multiple words.
```
Comment (kariabancroft) [inline views/home.erb:8]: This would be a nice place to include a link to these pages as well (even though I know they're also on the nav bar)

**Comment by kariabancroft** [comment]
Nice job with @shannachau! Your code is extremely readable and nicely laid out. It was nice a easy to read and figure out what you were trying to accomplish. I like seeing your commit history - you're definitely doing a good job of continuously committing when you update. Great job with your use of routes and HTTP verbs.

File: lib/score.rb (hunk 3 above)
Comment (carjug) [inline lib/score.rb:73]: I need to research more about blocks and procs. I think I get what you're saying -- we could define a proc where the code is repetitive to dry up the code more?

**Comment by carjug** [comment]
Thanks @kariabancroft <3

**Pull request merged**